### PR TITLE
[ci][train] Remove unnecessary `xgboost_ray`/`lightgbm_ray` reinstalls for release tests

### LINKED ISSUE #43155: Release test xgboost_tune_small.aws failed
Release test **xgboost_tune_small.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a4-c2a8-4382-b6c0-0c7d0a827ff2 for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43156: Release test lightgbm_tune_small.aws failed
Release test **lightgbm_tune_small.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a5-63d0-41fe-bab7-9c88f5dee58c for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43157: Release test lightgbm_tune_16x4.aws failed
Release test **lightgbm_tune_16x4.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a5-4457-4c9f-8742-d5de5b160212 for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43158: Release test lightgbm_tune_4x16.aws failed
Release test **lightgbm_tune_4x16.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a4-f338-4ada-bb58-60dc06e3af17 for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43159: Release test xgboost_tune_32x4.aws failed
Release test **xgboost_tune_32x4.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a4-daf8-4d3e-bfad-5480f3fca875 for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43161: Release test xgboost_tune_4x32.aws failed
Release test **xgboost_tune_4x32.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a4-ccd0-4ac7-a22a-e1719850b25c for more details.

Managed by OSS Test Policy

### LINKED ISSUE #43163: Release test air_benchmark_xgboost_cpu_10.aws failed
Release test **air_benchmark_xgboost_cpu_10.aws** failed. See https://buildkite.com/ray-project/release/builds/8408#018da6a5-4d08-440f-a929-47bb9aebed3f for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

https://github.com/ray-project/ray/pull/42111 removed classes that were depended on by the latest release of `xgboost_ray`. The master branch of this external library has some changes that remove this dependency, which was updated in `ml/core-requirements.txt`.

However, post build scripts for release tests install the latest version of these libraries, causing all these release test failures. We should test with the master version of the external libraries. This is a temporary measure, since we are planning to remove `xgboost_ray` and `lightgbm_ray` dependencies from Ray for 2.10 (to finish within 2 weeks).

## Related issue number

Closes https://github.com/ray-project/ray/issues/43155
Closes https://github.com/ray-project/ray/issues/43161
Closes https://github.com/ray-project/ray/issues/43163
Closes https://github.com/ray-project/ray/issues/43156
Closes https://github.com/ray-project/ray/issues/43157
Closes https://github.com/ray-project/ray/issues/43158
Closes https://github.com/ray-project/ray/issues/43159

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/byod/byod_xgboost_cuj_test.sh [hunk 1]
```diff
@@ -4,4 +4,4 @@
 
 set -exo pipefail
 
-pip3 install -U --force-reinstall --no-deps xgboost_ray
+pip install -U "git+https://github.com/ray-project/xgboost_ray.git"
```
Comment (aslonnie) [inline release/ray_release/byod/byod_xgboost_cuj_test.sh:7]: can we pin a version (tag or commit) on these pip installs?

**Comment by justinvyu** [comment]
@aslonnie Addressed your comment. Could I get another review / approval? This is to fix broken release tests.

**Review by aslonnie — APPROVED** [review]
this also works.

I thought you wanted to update the deps in `ml/core-requirements`

**Comment by justinvyu** [comment]
`ml/core-requirements` is not included in these release tests for some reason. See my attempt here: https://buildkite.com/ray-project/release/builds/8456#018da94d-2df2-4022-a383-63293e561427

I'm not going to spend time to figure it out since I am trying to remove this dependency within a week.

**Pull request merged**